### PR TITLE
Notification settings view model

### PR DESCRIPTION
**Pull request by juanlaube**
## What's this?
Adds a new notification settings view model, and an auxiliary generic view model to select an item from a list

### Relationships

Closes # 3127

## How is it done?
It's a new view model, very similar to the calendar settings view model, the new thing is the `SelectFromListViewModel` which can be reused everywhere we need to pick an item from a list

### Why not another way?
The `SelectFromListViewModel` could be something specific for the upcoming events config, but that's boring 😄 

### Side effects
None

## Review hints
Go commit by commit

## :squid: Permissions
Only myself

### CODE REVIEW

File: Toggl.Foundation.MvvmCross/Parameters/SelectFromListParameters.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        public string Title { get; set; }
+
+        public TValue Value { get; set; }
+    }
+
+    public struct SelectFromListParameters<TValue>
+    {
+        public IList<SelectableItem<TValue>> Items { get; set; }
+
+        public int SelectedIndex { get; set; }
```
Comment (sofokli1) [inline Toggl.Foundation.MvvmCross/Parameters/SelectFromListParameters.cs]: Why not make this usable without a default selection? Ex: when you want to make the user make a choice, when there is no default choice.

File: Toggl.Foundation.MvvmCross/ViewModels/Settings/NotificationSettingsViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                saveUpcomingEventsOption(selection);
+            }).SelectUnit();
+        }
+
+        private IObservable<int> currentUpcomingEventsOptionIndex()
+        {
+            return Observable.FromAsync(async () =>
+            {
+                var notificationsEnabled = await userPreferences.CalendarNotificationsEnabled;
+                var timespan = await userPreferences.TimeSpanBeforeCalendarNotifications;
```
Comment (sofokli1) [inline Toggl.Foundation.MvvmCross/ViewModels/Settings/NotificationSettingsViewModel.cs]: Shouldn't `timespan` be `timeSpan`?

File: Toggl.Foundation.MvvmCross/Parameters/SelectFromListParameters.cs (hunk 1 above)
Comment (juanlaube) [reply Toggl.Foundation.MvvmCross/Parameters/SelectFromListParameters.cs]: That's a good idea, but I will let someone who needs that particular use case to do it 😂

File: Toggl.Foundation.MvvmCross/ViewModels/Settings/NotificationSettingsViewModel.cs (hunk 2 above)
Comment (juanlaube) [reply Toggl.Foundation.MvvmCross/ViewModels/Settings/NotificationSettingsViewModel.cs]: Good catch

**Pull request merged**